hack/lib/golang.sh

```diff
@@ -554,6 +554,18 @@ kube::golang::setup_env() {
 
   # This seems to matter to some tools
   export GO15VENDOREXPERIMENT=1
+
+  # GOMAXPROCS by default does not reflect the number of cpu(s) available
+  # when running in a container, please see https://github.com/golang/go/issues/33803
+  if ! command -v ncpu >/dev/null 2>&1; then
+    kube::log::status "ncpu not found; installing from hack/tools"
+    pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+    GO111MODULE=on go install ./ncpu
```

Comment:
We can use something like `go build -o ${KUBE_ROOT}/_output/ncpu` to avoid polluting the system path, and if we DRY out that location we can still check if it exists before building.

like https://github.com/kubernetes/kubernetes/blob/c3e7eca7fd38454200819b60e58144d5727f1bbc/hack/verify-golangci-lint.sh#L132

In some other places we're able to rely on GOBIN but we've just unset it above and haven't set it here

Reply:
I'm not sure what the most appropriate path is, but something other than system bin

Reply:
@BenTheElder you run and check, it will not pollute. i can see `./_output/dockerized/bin/linux/arm64/ncpu`